Sources/Swift/Integrations/SessionReplay/SentryOnDemandReplay.swift

```diff
@@ -0,0 +1,164 @@
+#if canImport(UIKit) && !SENTRY_NO_UIKIT
+#if os(iOS) || os(tvOS)
+
+@_implementationOnly import _SentryPrivate
+import AVFoundation
+import CoreGraphics
+import Foundation
+import UIKit
+
+struct SentryReplayFrame {
+    let imagePath: String
+    let time: Date
+    
+    init(imagePath: String, time: Date) {
+        self.imagePath = imagePath
+        self.time = time
+    }
+}
+
+@available(iOS 16.0, tvOS 16.0, *)
+@objcMembers
+class SentryOnDemandReplay: NSObject {
+    private let _outputPath: String
+    private let _onDemandDispatchQueue: DispatchQueue
+    
+    private var _starttime = Date()
+    private var _frames = [SentryReplayFrame]()
+    private var _currentPixelBuffer: SentryPixelBuffer?
+    
+    var videoSize = CGSize(width: 200, height: 434)
+    var bitRate = 20_000
+    var frameRate = 1
+    var cacheMaxSize = UInt.max
+    
+    init(outputPath: String) {
+        self._outputPath = outputPath
+        _onDemandDispatchQueue = DispatchQueue(label: "io.sentry.sessionreplay.ondemand")
+    }
+    
+    func addFrame(image: UIImage) {
+        _onDemandDispatchQueue.async {
+            self.asyncAddFrame(image: image)
+        }
+    }
+    
+    private func asyncAddFrame(image: UIImage) {
+        guard let data = resizeImage(image, maxWidth: 300)?.pngData() else { return }
+        
+        let date = Date()
+        let interval = date.timeIntervalSince(_starttime)
+        let imagePath = (_outputPath as NSString).appendingPathComponent("\(interval).png")
+        try? data.write(to: URL(fileURLWithPath: imagePath))
+        _frames.append(SentryReplayFrame(imagePath: imagePath, time: date))
+        
+        while _frames.count > cacheMaxSize {
+            let first = _frames.removeFirst()
+            try? FileManager.default.removeItem(at: URL(fileURLWithPath: first.imagePath))
+        }
+    }
```

Comment:
For Flutter, I think it would be useful to expose `asyncAddFrame(imagePath: NSString, date: Date)`

My idea is to:
- crate a SentryOnDemandReplay from a Flutter swift binding code
- create screenshots in Flutter
- write pngs to the cache directory directly in Flutter
- call asyncAddFrame(filename)
- call createVideoWith() to get the video file path
- send the replay envelope from Flutter

Reply:
@brustolin WDYT?

Reply:
Yeah, if its useful to Flutter we can make it public.